src/main/java/com/baidu/hugegraph/rpc/RpcServer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 HugeGraph Authors
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.baidu.hugegraph.rpc;
+
+import java.util.Map;
+
+import org.apache.commons.collections.MapUtils;
+import org.slf4j.Logger;
+
+import com.alipay.sofa.rpc.config.ProviderConfig;
+import com.alipay.sofa.rpc.config.ServerConfig;
+import com.baidu.hugegraph.config.HugeConfig;
+import com.baidu.hugegraph.config.RpcOptions;
+import com.baidu.hugegraph.util.Log;
+
+public class RpcServer {
+
+    private static final Logger LOG = Log.logger(RpcServer.class);
+
+    private final HugeConfig conf;
+    private final RpcProviderConfig configs;
+    private final ServerConfig serverConfig;
+
+    public RpcServer(HugeConfig conf) {
+        RpcCommonConfig.initRpcConfigs(conf);
+        this.conf = conf;
+        this.serverConfig = new ServerConfig();
+        this.serverConfig.setProtocol(conf.get(RpcOptions.RPC_PROTOCOL))
+                         .setHost(conf.get(RpcOptions.RPC_SERVER_HOST))
+                         .setPort(conf.get(RpcOptions.RPC_SERVER_PORT))
+                         .setDaemon(false);
```

Comment:
How to listen on a random port? There are a few of workers on a computer, and they can't listen on same port.

Reply:
maybe we can use sofa  option`server.adaptive.port`